### DISCUSSION TITLE
Flutter Auth Session

### DISCUSSION THREAD

**Discussion by ihadabs** [supabase/supabase#5461] [Questions] [answered]
Hey there,

Im working on a Flutetr app and using Supabase authentication.
I am having a hard time with keeping users logged in.
Im calling `recoverSupabaseSession()` on Splash screen but still users has to login once or twice a day.

How can I tell Supabase client to keep the session for longer time on the user device?

Firebase keeps the session for a very long time, i guess indefinitely persisted according to their docs.
It even persist the session on uninstalls and re-installs.
https://firebase.google.com/docs/auth/web/auth-state-persistence#web-version-9

Thanks.

**Comment by dshukertjr** [comment, marked as answer]
Hi @ihadabs !

Are you using [supabase](https://pub.dev/packages/supabase) package or the [supabase_flutter](https://pub.dev/packages/supabase_flutter) package for your project? 

`supabase_flutter` package has built in mechanism to keep the session alive without manually calling `recoverSupabaseSession()`, so if you are currently using `supabase` package, you might want to consider migrating to `supabase_flutter` package. 

You can find an example app [here](https://supabase.com/docs/guides/with-flutter)!

**Reply by ihadabs** [reply]
Thanks for replying @dshukertjr,
I have done exactly as the example on Supabase docs, they are also using  `recoverSupabaseSession()`.

But for some reason its not working...

**Reply by dshukertjr** [reply]
Hi @ihadabs ! Sorry for the late response, and thank you for more info. 

So you are currently using supabase_flutter package, and followed the [quick starter guide](https://supabase.com/docs/guides/with-flutter), but still the auth state is not persisted? If so, which version of supabase_flutter are you using?

**Reply by ihadabs** [reply]
Hey @dshukertjr,
I contacted the support and I was able to make the session presist for 1 week max from Supabase dashboard.

The refresh_token is not yet used in the package to presist the session for longer (refresh the token before its expired).

Thanks
